### ISSUE TITLE
Increase task expiration limits

### ISSUE THREAD

**Issue by eu9ene** [mozilla/firefox-translations-training#153]
Model training can take weeks (depending on the hardware, dataset size and hyperparameters), so we should increase the limit from 24 hours to, say, a month

**Comment by gabrielBusta** [comment]
taskcluster/taskcluster#6269

**Comment by petemoore** [comment]
Note, workers run on spot instances, which may be terminated at any time. Please be aware once we increase the max run time of a task, there is still the possibility that a worker node is terminated at any time.

If there is any way to dissect the model training into multiple discrete tasks (even if they are sequential) this would help recovery in the case of a spot termination. It is highly possible that weeks into a task, we may lose a node through a spot termination, and I would hate all the training and compute time invested to be lost.

Is there some way to snapshot state e.g. after an hour, and then have a subsequent task that loads this state (e.g. by mounting task artifacts) and then continues training? It feels like having the expectation that a task can last multiple days is going to lead to an inevitable situation where at some point a spot termination is going to seriously set us back in time.

**Comment by eu9ene** [comment]
>Is there some way to snapshot state e.g. after an hour, and then have a subsequent task that loads this state (e.g. by mounting task artifacts) and then continues training?

Yes, and continuation of training should be the default behaviour for all our training steps if they fail/cancelled. Marian saves checkpoints, we just need to connect the retry/rerun to the same artifacts. If the task retries, it should load the last marian checkpoint. We should still remove the limits not to deal with constant task restarts in the regular training mode. see https://github.com/mozilla/firefox-translations-training/issues/154

**Comment by gabrielBusta** [comment]
We definitely need to implement a way to continue the training. I will start looking into that.

I think it makes sense to configure the maximum run time allowed per worker-pool because it depends on where the workers are running. In the case of our GCP spot VMs there's no pre-defined maximum run time (and we can set a maximum run time as high as 120 days.)

But, I wonder if this limit is bound limits on the queue entries? The dependencies on our long running tasks will exceeded the deadline

**Comment by gabrielBusta** [comment]
taskcluster/taskcluster#2942
